sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/PartitionMetadataCache.scala

```diff
@@ -176,27 +243,70 @@ private object PartitionMetadataCache extends BasicLoggingTrait {
       .to(clientCacheItem => {
         val container = ThroughputControlHelper.getContainer(userConfig, cosmosContainerConfig, clientCacheItem.client)
 
-        val options = CosmosChangeFeedRequestOptions.createForProcessingFromNow(
+        val optionsFromNow = CosmosChangeFeedRequestOptions.createForProcessingFromNow(
           SparkBridgeImplementationInternal.toFeedRange(feedRange))
-        options.setMaxItemCount(1)
-        options.setMaxPrefetchPageCount(1)
-        options.setQuotaInfoEnabled(true)
+        optionsFromNow.setMaxItemCount(1)
+        optionsFromNow.setMaxPrefetchPageCount(1)
+        optionsFromNow.setQuotaInfoEnabled(true)
 
         val lastDocumentCount = new AtomicLong()
         val lastTotalDocumentSize = new AtomicLong()
         val lastContinuationToken = new AtomicReference[String]()
+        val firstLsn = new AtomicLong(-1)
 
-        container
-          .queryChangeFeed(options, classOf[ObjectNode])
-          .handle(r => {
+        val fromNowFlux: Flux[Int] = container
```

Comment:
Should we first do the From Beginning and only if that one has a none-0 LSN, then do the second From Now? Otherwise for an empty FeedRange we are doing a second unnecessary CF request, right?

Reply:
True - but optimizing for empty feed ranges isn't really worth it :-)

Reply:
Or maybe it's an unnecessary optimization